alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunnerService.kt

```diff
@@ -220,9 +230,7 @@ object MonitorRunnerService : JobRunner, CoroutineScope, AbstractLifecycleCompon
         launch {
             try {
                 monitorCtx.moveAlertsRetryPolicy!!.retry(logger) {
-                    if (monitorCtx.alertIndices!!.isAlertInitialized()) {
-                        moveAlerts(monitorCtx.client!!, jobId, null)
-                    }
+                    moveAlerts(monitorCtx.client!!, jobId, null, monitorCtx)
```

Comment:
Is this version of the function only for workflows? If so, we need the other one for monitors.

Reply:
calling both functions because after deletion we dont know from job id if its a monitor or workflow(we can create an issue to create id with monitor/workflow prefix)

Reply:
We need to check if the alerts index is initialized if its not using a different datasource.